modules/test-operators/helpers/memcached.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2023 Red Hat
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"fmt"
+
+	t "github.com/onsi/gomega"
+	memcachedv1 "github.com/openstack-k8s-operators/infra-operator/apis/memcached/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// CreateMemcached creates a new Memcached instance with the specified namespace in the Kubernetes cluster.
+func (tc *TestHelper) CreateMemcached(namespace string, memcachedName string, spec memcachedv1.MemcachedSpec) types.NamespacedName {
+	name := types.NamespacedName{
+		Name:      memcachedName,
+		Namespace: namespace,
+	}
+
+	mc := &memcachedv1.Memcached{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "memcached.openstack.org/v1beta1",
+			Kind:       "Memcached",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      memcachedName,
+			Namespace: namespace,
+		},
+		Spec: spec,
+	}
+
+	t.Expect(tc.K8sClient.Create(tc.Ctx, mc)).Should(t.Succeed())
+
+	return name
```

Comment:
Probably a remainder of an older pattern where only the namespace was passed in and the name was generated by call itself.

Reply:
my point was more, we pass in the name and namespace. which is 1:1 set in the namespaced name. so this seem to be not really required to return it. the caller already has all the details. but I am fine with it if it is the pattern we also use in other places

Reply:
> Probably a remainder of an older pattern where only the namespace was passed in and the name was generated by call itself.

yes probably

Reply:
returning the NamespacedName might be redundant, but would allow you to call DeferCleanup directl like `DeferCleanup(th.DeleteInstance, CreateMemcached(...))` or `DeferCleanup(th.DeleteMemcached, CreateMemcached(...))` and we use this pattern quite often. So allowing this interface would be less confusing.